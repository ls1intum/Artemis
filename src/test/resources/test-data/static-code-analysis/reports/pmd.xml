<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd" version="6.33.0" timestamp="2021-07-04T16:55:29.853">
    <file name="/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java">
        <violation beginline="6" endline="6" begincolumn="1" endcolumn="26" rule="UnusedImports" ruleset="Best Practices" package="code" class="Client" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_bestpractices.html#unusedimports" priority="4">
            Avoid unused imports such as 'javax.swing.JFrame'
        </violation>
        <violation beginline="7" endline="7" begincolumn="1" endcolumn="19" rule="DontImportJavaLang" ruleset="Code Style" package="code" class="Client" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_codestyle.html#dontimportjavalang" priority="4">
            Avoid importing anything from the package java.lang
        </violation>
        <violation beginline="7" endline="7" begincolumn="1" endcolumn="19" rule="UnusedImports" ruleset="Best Practices" package="code" class="Client" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_bestpractices.html#unusedimports" priority="4">
            Avoid unused imports such as 'java.lang'
        </violation>
        <violation beginline="73" endline="73" begincolumn="18" endcolumn="27" rule="UnusedLocalVariable" ruleset="Best Practices" package="code" class="Client" method="createRandomDatesList" variable="randomDate" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_bestpractices.html#unusedlocalvariable" priority="3">
            Avoid unused local variables such as 'randomDate'.
        </violation>
        <violation beginline="81" endline="81" begincolumn="31" endcolumn="53" rule="UnusedPrivateMethod" ruleset="Best Practices" package="code" class="Client" method="CreateRandomDatesList" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_bestpractices.html#unusedprivatemethod" priority="3">
            Avoid unused private methods such as 'CreateRandomDatesList()'.
        </violation>
        <violation beginline="94" endline="94" begincolumn="9" endcolumn="9" rule="EmptyStatementNotInLoop" ruleset="Error Prone" package="code" class="Client" method="CreateRandomDatesList" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_errorprone.html#emptystatementnotinloop" priority="3">
            An empty statement (semicolon) not part of a loop
        </violation>
        <violation beginline="94" endline="94" begincolumn="10" endcolumn="10" rule="EmptyStatementNotInLoop" ruleset="Error Prone" package="code" class="Client" method="CreateRandomDatesList" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_errorprone.html#emptystatementnotinloop" priority="3">
            An empty statement (semicolon) not part of a loop
        </violation>
        <violation beginline="94" endline="94" begincolumn="11" endcolumn="11" rule="EmptyStatementNotInLoop" ruleset="Error Prone" package="code" class="Client" method="CreateRandomDatesList" externalInfoUrl="https://pmd.github.io/pmd-6.33.0/pmd_rules_java_errorprone.html#emptystatementnotinloop" priority="3">
            An empty statement (semicolon) not part of a loop
        </violation>
    </file>
</pmd>
