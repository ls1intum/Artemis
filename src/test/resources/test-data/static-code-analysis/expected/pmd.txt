{"tool":"PMD","issues":[{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":6,"endLine":6,"startColumn":1,"endColumn":26,"rule":"UnusedImports","category":"Best Practices","message":"Avoid unused imports such as 'javax.swing.JFrame'","priority":"4"},{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":7,"endLine":7,"startColumn":1,"endColumn":19,"rule":"DontImportJavaLang","category":"Code Style","message":"Avoid importing anything from the package java.lang","priority":"4"},{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":7,"endLine":7,"startColumn":1,"endColumn":19,"rule":"UnusedImports","category":"Best Practices","message":"Avoid unused imports such as 'java.lang'","priority":"4"},{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":73,"endLine":73,"startColumn":18,"endColumn":27,"rule":"UnusedLocalVariable","category":"Best Practices","message":"Avoid unused local variables such as 'randomDate'.","priority":"3"},{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":81,"endLine":81,"startColumn":31,"endColumn":53,"rule":"UnusedPrivateMethod","category":"Best Practices","message":"Avoid unused private methods such as 'CreateRandomDatesList()'.","priority":"3"},{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":94,"endLine":94,"startColumn":9,"endColumn":9,"rule":"EmptyStatementNotInLoop","category":"Error Prone","message":"An empty statement (semicolon) not part of a loop","priority":"3"},{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":94,"endLine":94,"startColumn":10,"endColumn":10,"rule":"EmptyStatementNotInLoop","category":"Error Prone","message":"An empty statement (semicolon) not part of a loop","priority":"3"},{"filePath":"/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/MDTESTCODE-ARTEMISTESTUSER1-JOB1/assignment/src/code/Client.java","startLine":94,"endLine":94,"startColumn":11,"endColumn":11,"rule":"EmptyStatementNotInLoop","category":"Error Prone","message":"An empty statement (semicolon) not part of a loop","priority":"3"}]}
